File: POO/TP2/Practica Python/EJ5/Persona.py

from datetime import date
#date.today()

class Persona:
    __nombre: str
    __apellido: str
    __fecha_nacimiento = date
    
    def __init__(self, nombre, apellido, fecha_nacimiento):
        self.__nombre= nombre
        self.__apellido = apellido
        fecha_parseada = date.fromisoformat(fecha_nacimiento)
        self.__fecha_nacimiento = fecha_parseada

    def get_nombre(self):
        return self.__nombre
    
    def set_nombre(self, nombre):
        self.__nombre = nombre
    
    def get_apellido(self):
        return self.__apellido
    
    def set_apellido(self, apellido):
        self.__apellido = apellido
    
    def get_fecha_nacimiento(self):
        return self.__fecha_nacimiento
    
    def set_fecha_nacimiento(self, fecha_nacimiento):
        fecha_parseada = date.fromisoformat(fecha_nacimiento)
        self.__fecha_nacimiento = fecha_parseada
    
    def toString(self):
        return f'Nombre: {self.__nombre}, Apellido: {self.__apellido},\n Fecha de nacimiento: {self.__fecha_nacimiento}\n ' 

File: POO/TP2/Practica Java/EJ5/Persona.java
package EJ5;

import java.time.LocalDate;

public class Persona {
    private String nombre;
    private String apellido;
    private LocalDate fechaNacimiento;

    public Persona(String nombre, String apellido, String fechaNacimiento) {
        this.nombre = nombre;
        this.apellido = apellido;
        this.fechaNacimiento = LocalDate.parse(fechaNacimiento);
    }

    public String getNombre() {
        return this.nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getApellido() {
        return this.apellido;
    }

    public void setApellido(String apellido) {
        this.apellido = apellido;
    }

    public LocalDate getFechaNacimiento() {
        return this.fechaNacimiento;
    }

    public void setFechaNacimiento(LocalDate fechaNacimiento) {
        this.fechaNacimiento = fechaNacimiento;
    }

    public String toString() {
        return "Nombre: " + this.nombre + ", Apellido: " + this.apellido + ", Fecha de Nacimiento: "
                + this.fechaNacimiento;
    }

}
